### PR TITLE
support firebase-functions 7.x.x

### PR DESCRIPTION
**Pull request by Tkkg1994**
## Summary by CodeRabbit

* **Chores**
  * Updated Firebase Functions dependency to v7.0.0
  * Expanded Firebase Functions version compatibility in Nuxt package to include v7.0.0
  * Updated CI workflow to use Java 21

* **Tests**
  * Refactored Firestore collection tests for improved test isolation

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by Tkkg1994** [comment]
@posva 
Thanks for all the help, I closed the PR https://github.com/vuejs/vuefire/pull/1647 and now opened this one with the 3 correct commits.

Hopefully this should go through.

Wish you a wonderful rest of the 2025!

**Comment by Tkkg1994** [comment]
Fixes https://github.com/vuejs/vuefire/issues/1646

**Comment by posva** [comment]
Thanks a lot!

Now you should be able to see a published version at https://github.com/vuejs/vuefire/actions/runs/20583068403/job/59115705184?pr=1648 and [in a comment here](https://github.com/vuejs/vuefire/pull/1648#issuecomment-3697620588)

I don't know when this will go forward, I still have to check with firebase and unfortunately, I currently have a lot of other pressing matters, so it will be probably months before I can merge this. Until that, you can use this PR's version

**Comment by Tkkg1994** [comment]
Perfect, this looks really nice @posva 

Anything I can do to speed the process up? I have hoped to be able to install an official version soon. I can also be helpful to assist as I have a personal interest and maybe some more time than you 😄 

Greetings

Edit: and I would need the nuxt-module, so nuxt-vuefire

**Comment by posva** [comment]
Nothing you can do on your side, thanks!
Send another PR and mention me so I can approve the workflow and it publishes

**Comment by Tkkg1994** [comment]
Thank you for the answer @posva 

As far as I can see I already included the correct changes here in https://github.com/vuejs/vuefire/pull/1648/changes/9f0699242a0963b190a06d166b9d24854108ba96#diff-36b207d79434b922b863a3e8d4ca66d566cf0044d5f4d0943b208f7c19722dc2

So basically we should be able to build this without any further PR

File: tests/firestore/collection.spec.ts [hunk 1]
```diff
@@ -256,7 +256,11 @@ describe(
       expect(data.value).toEqual(copy)
     })
 
-    it('rejects on error', async () => {
+    // Collections don't naturally produce subscription errors in the emulator like documents do.
```
Comment (posva) [inline tests/firestore/collection.spec.ts:259]: Ideally, we want to find a test replacement for this PR to get merged

**Comment by brambekkers** [comment]
Can we please merge this pr? It would really help since i can only install packages that are autorized by my organization and they only do that with official releases

File: tests/firestore/collection.spec.ts (hunk 1 above)
Comment (Tkkg1994) [reply tests/firestore/collection.spec.ts:259]: @posva any suggestion on how to implement this? In this case, I could help to also include this test with the latest  7.xx functions

**Comment by Tkkg1994** [comment]
Dear @posva 

I truly want to help.

Since I can't yet install this module (including the nuxt-version of it), what can I do to support a faster realease. Let me know, I will be here to help!

Greetings 😄

**Comment by posva** [comment]
@Tkkg1994 I'm currently busy with other work. I can make an exception and prioritize this work like [I have done before](https://cal.com/posva/consultancy?month=2026-03) if I get paid for it but since I disabled cal.com, [sponsor $250 me one time instead](https://github.com/sponsors/posva/sponsorships?sponsor=posva&frequency=one-time&amount=250) and I will merge and release this

**Comment by Tkkg1994** [comment]
Hi @posva,

Thanks for the transparency about your current workload — I completely get that maintaining popular packages like VueFire (plus everything else in the Vue ecosystem) is a huge effort, and time is limited.

I truly appreciate all the work you've put into this library over the years — it's been incredibly valuable to many of us.

That said, I did already invest quite a bit of my own time researching, implementing, testing, and iterating on this PR (including the dependency bump, CI fix, test refactor, and addressing the CodeRabbit feedback). So while I understand you sometimes need to prioritize paid/sponsored work to keep things sustainable, it doesn't feel quite fair to me personally to now pay extra on top of that contribution just to get it merged and released.

I'm more than happy to keep helping reduce your load for free where I can:
- Implement a replacement for the skipped "rejects on error" test (if you have a preferred approach/mock strategy, just point me in the right direction and I'll take a shot at it).
- Write/update docs or examples for the v7 support once it's merged.
- Test edge cases or help verify other related PRs.
- Rally more users who need this (e.g. ping others blocked by #1646 in discussions/forums).

If enough people chime in showing strong need (businesses/teams blocked on official releases due to policies), would that help bump it up the queue for a minimal merge + release? Or is there another low-friction way forward that doesn't involve the sponsorship exception?

No pressure either way — just wanted to be honest about where I'm at while offering what I can contribute.

Thanks again for everything ❤️

**Comment by posva** [comment]
@Tkkg1994 Thanks again for the work you put into this PR, I really appreciate it 🙏

The release process isn’t just a quick merge; it takes a couple of hours to properly verify everything and publish safely. Right now I need to prioritize other things.

If a company depends on an official release due to internal policies, a one-time $250 sponsorship allows me to allocate the time for it. Otherwise, I’ll get to it when I can.

For those who don’t require an official release, **you can already publish and use your own version from the PR thanks to the automated release setup.**

I’ll also reach out to Google to see if there’s interest in supporting the ongoing maintenance.

**Comment by Tkkg1994** [comment]
@posva 
I understand that 100%

Did you hear anything from Google? Else we can start a fund me page, maybe some people would contribute. It's easier than one person paying the whole amount

**Comment by Tkkg1994** [comment]
Dear @posva 

I wish you a very happy easter! Hope everything's fine. Wanted to ask how this is going and any news of funding/go fund me 😄 

Also, when I could get the money, how fast could you release an update to vuefire and nuxt-vuefire?

Greetings

**Comment by Tkkg1994** [comment]
Hello dear @posva 

In case you could promise me to merge this request and do a release for both vuefire and nuxt vuefire in the upcoming week (from the 13th of april) I'll be able to send you the 250 dollars.

It's starting to hurt me😅

**Comment by Tkkg1994** [comment]
Dear @posva 

I can confirm that I just sponsored 250 dollars via Github specifically for this project.

Thanks for the great work!

Greetings

FYI @brambekkers
